### PR TITLE
ARROW-13402: [R] Update flight.Rmd vignette

### PR DESCRIPTION
**Pull request by thisisnic**

### CODE REVIEW

**Comment by thisisnic** [comment]
I've made some minor, mainly stylistic changes in this PR, but still feel that this vignette is a little short.  I was thinking about adding a little bit of information on why you might want to use Flight, but didn't know if that would fit well here.  If anyone has any suggestions, let me know.

File: r/vignettes/flight.Rmd [hunk 1]
```diff
@@ -74,5 +82,6 @@ client %>%
 ## 3 virginica        6.9
 ```
 
-Because `flight_get()` returns an Arrow data structure, we can directly pipe
-its result into a `dplyr` workflow.
+Because `flight_get()` returns an Arrow data structure, you can directly pipe
+its result into a [dplyr](https://dplyr.tidyverse.org/) workflow.
+See [the dataset vignette](./dataset.html) for more information on working with Arrow objects via a dplyr interface.
```
Comment (nealrichardson) [inline r/vignettes/flight.Rmd:87]: pkgdown will linkify this correctly and it should work around any quirks in how CRAN makes URLs

```suggestion
See `vignette("dataset", package = "arrow")` for more information on working with Arrow objects via a dplyr interface.
```

**Review by nealrichardson — APPROVED** [review]
One suggestion but LGTM

File: r/vignettes/flight.Rmd (hunk 1 above)
Comment (thisisnic) [reply r/vignettes/flight.Rmd:87]: Thanks!

**Pull request closed without merging**